gateway/mw_basic_auth.go

```diff
@@ -178,7 +178,7 @@ func (k *BasicAuthKeyIsValid) ProcessRequest(w http.ResponseWriter, r *http.Requ
 	}
 
 	// Check if API key valid
-	keyName := generateToken(k.Spec.OrgID, username)
+	keyName := username
```

Comment:
@buger As hashed part is shrunk, collisions occur and the `TestBasicAuthCachedUserCollision` is failing. I think it is expected.

Reply:
A possible solution is to double the key to be cached:
```
	cacheKey := key
	if t.Spec.GlobalConfig.HashKeys {
		cacheKey = storage.HashStr(key `+key`)
	}
```